tests/dataframe/test_fill.py

```diff
@@ -0,0 +1,24 @@
+from __future__ import annotations
+
+from typing import Any
+
+import pytest
+
+import daft.errors
+from daft import DataFrame
+
+
+def test_fill_nan_data_frame(missing_value_data: list[dict[str, Any]]) -> None:
+    df = DataFrame.from_pylist(missing_value_data)
+    df = df.with_column("sepal_width", df["sepal_width"].fillnan(3.1))
+    with pytest.raises(daft.errors.ExpressionTypeError):
+        df = df.with_column("sepal_width_none", df["sepal_width"].fillnan(None))
```

Comment:
@jaychia None can´t be used with float columns, since it throws an expressiontypeerror. In Pandas None will be converted to NaN if the column is float. 
Do you know how I am able to check the type of a column to assert that this is only applied to float columns? There is the method _resolve_type, but one needs the schema for this. I was thinking to supply the schema if an expression is derived by DataFrame.__getitem__ but that wouldn't solve it for col(column_name) calls.

Reply:
Ah indeed! This seems to be a corner-case bug that we haven't anticipated I think.

Under the hood `lit(None)` gets a `Null` type, but it should actually take on whatever type it is applied against.

Feel free to add a comment/skip this test for now - in our upcoming Rust release, these will be handled appropriately through type-promotion (essentially, `float + null = float`)

Reply:
For the second part of your question about asserting that it runs against a float column, this should actually already be handled by your use of `.is_nan()`.

For example, the following snippet will fail:

```
df = DataFrame.from_pydict({"x": ["a", "b", "c"]})

# Throws ExpressionTypeError: Unable to resolve type for operation: is_nan(col(x))
df.with_column("should_fail", df["x"].is_nan())
```